### PR TITLE
Entity linking

### PR DESCRIPTION
**Pull request by dmitrijeuseew**
Config and files for entity linking

### CODE REVIEW

File: deeppavlov/models/kbqa/entity_linking.py [hunk 1]
```diff
@@ -13,317 +13,312 @@
 # limitations under the License.
 
 import re
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:15]: redundant

File: deeppavlov/models/kbqa/entity_linking.py [hunk 2]
```diff
@@ -13,317 +13,312 @@
...
-import sqlite3
 from logging import getLogger
 from typing import List, Dict, Tuple, Optional, Any
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:17]: Optional, Any are redundant

File: deeppavlov/models/kbqa/entity_linking.py [hunk 3]
```diff
@@ -13,317 +13,312 @@
...
-from collections import defaultdict
+from collections import defaultdict, Counter
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:18]: Counter is redundant

File: deeppavlov/models/kbqa/entity_linking.py [hunk 4]
```diff
@@ -13,317 +13,312 @@
...
 
+import numpy as np
 import nltk
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:21]: redundant

File: docs/features/models/entity_linking.rst [hunk 5]
```diff
@@ -0,0 +1,41 @@
+Entity Linking
+========================================
+
+Entity linking is the task of mapping of words in the text, corresponding to names of persons, locations and organizations, to entities in the target knowledge base (Wikidata in our case).
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:4]:
```suggestion
Entity linking is the task of mapping words from text (e.g. names of persons, locations and organizations) to entities from the target knowledge base (Wikidata in our case).
```

File: deeppavlov/configs/kbqa/entity_linking_rus.json [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      "ROOT_PATH": "~/.deeppavlov",
+      "DOWNLOADS_PATH": "{ROOT_PATH}/downloads",
+      "MODELS_PATH": "{ROOT_PATH}/models",
+      "CONFIGS_PATH": "{DEEPPAVLOV_PATH}/configs",
+      "NER_PATH": "{MODELS_PATH}/ner_rus_bert"
+    },
+    "requirements": [
+      "{DEEPPAVLOV_PATH}/requirements/tf.txt",
+      "{DEEPPAVLOV_PATH}/requirements/bert_dp.txt",
+      "{DEEPPAVLOV_PATH}/requirements/faiss.txt
```
Comment (IgnatovFedor) [inline deeppavlov/configs/kbqa/entity_linking_rus.json]:
```suggestion
      "{DEEPPAVLOV_PATH}/requirements/faiss.txt"
```

File: deeppavlov/requirements/faiss.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+faiss==1.6.3
```
Comment (IgnatovFedor) [inline deeppavlov/requirements/faiss.txt:1]: There is no faiss 1.6.3 in PyPI. Eather you mean faiss-cpu==1.6.3, or faiss==1.5.3

File: docs/features/models/entity_linking.rst [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+.. code:: bash
+
+    python -m deeppavlov install entity_linking_rus
+
+To use a pre-trained model from CLI use the following command:
+
+.. code:: bash
+
+    python deeppavlov/deep.py interact entity_linking_rus [-d]
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:26]:
```suggestion
    python -m deeppavlov interact entity_linking_rus -d
```

File: deeppavlov/configs/kbqa/entity_linking_rus.json [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  },
+  "metadata": {
+    "variables": {
+      "ROOT_PATH": "~/.deeppavlov",
+      "DOWNLOADS_PATH": "{ROOT_PATH}/downloads",
+      "MODELS_PATH": "{ROOT_PATH}/models",
+      "CONFIGS_PATH": "{DEEPPAVLOV_PATH}/configs",
+      "NER_PATH": "{MODELS_PATH}/ner_rus_bert"
+    },
+    "requirements": [
```
Comment (IgnatovFedor) [inline deeppavlov/configs/kbqa/entity_linking_rus.json:69]: Add hdt.txt to requirements

File: docs/features/models/entity_linking.rst [hunk 10]
```diff
@@ -0,0 +1,41 @@
+Entity Linking
+========================================
+
+Entity linking is the task of mapping of words in the text, corresponding to names of persons, locations and organizations, to entities in the target knowledge base (Wikidata in our case).
+
+Entity Linking component consists of the following steps:
+
+* the substring, detected with :config:`NER <ner/ner_rus_bert_probas.json>`, is fed into TfidfVectorizer with the following parameters: analyzer="char_wb", ngram_range=(2, 2), max_features=500, and the vector is converted to dense
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:8]:
```suggestion
* the substring, detected with :config:`NER <ner/ner_rus_bert_probas.json>`, is fed to TfidfVectorizer and the resulting sparse vector is converted to dense one
```

File: docs/features/models/entity_linking.rst [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+* `Faiss <https://github.com/facebookresearch/faiss>`__ library is used to find k nearest neighbours for tf-idf vector in the matrix where rows correspond to tf-idf vectors of words in entity titles
+* Entities are ranked by number of relations in Wikidata (number of outgoing edges of nodes in the knowledge graph)
+* :config:`BERT <classifiers/entity_ranking_bert_rus_no_mention.json>` is used for ranking of entities by description of the entity and the sentence with the mention of the entity.
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:11]:
```suggestion
* :config:`BERT <classifiers/entity_ranking_bert_rus_no_mention.json>` is used for entities ranking by entity description and the sentence with the mention of the entity.
```
And I don't quite understand what you mean by `and the sentence with the mention of the entity`

File: docs/features/models/entity_linking.rst [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+To use a pre-trained model from CLI use the following command:
+
+.. code:: bash
+
+    python deeppavlov/deep.py interact entity_linking_rus [-d]
+
+where ``entity_linking_rus`` and others are the names of configs and ``-d`` is an optional download key. The key ``-d`` is used
+to download the pre-trained model along with embeddings and all other files needed to run the model. Also command
+``download`` is possible,
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:30]: This information is unnecessary

File: docs/features/models/entity_linking.rst [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+
+Entity Linking model can be used from Python using the following code:
+
+.. code:: python
+
+    from deeppavlov import configs, build_model
+
+    el_model = build_model(configs.kbqa.entity_linking_rus, download=True)
+    el_model(['Москва — столица России, город федерального значения, административный центр Центрального федерального округа и центр Московской области.'])
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:40]: Add `>>>` if describing interactive mode

File: deeppavlov/models/kbqa/entity_detection_parser.py [hunk 14]
```diff
@@ -24,20 +26,41 @@
...
+            thres_proba: if the probability of the tag is less than thres_proba, we assign the tag as 'O'
+        """
+        self.entity_tags = entity_tags
+        self.type_tag = type_tag
+        self.o_tag = o_tag
         self.thres_proba = thres_proba
+        self.tag_ind_dict = {}
+        with open(str(expand_path(tags_file))) as fl:
+            tags = [line.split('\t')[0] for line in fl.readlines()]
+            self.entity_prob_ind = {entity_tag: [i for i, tag in enumerate(tags) if entity_tag in tag] for entity_tag in self.entity_tags}
```
[the remark this answers is not in the document]
Comment (dmitrijeuseew) [reply deeppavlov/models/kbqa/entity_detection_parser.py:46]: tags are ["O", "I-PER", "B-PER", ...], and entity_tags are ["[PER", "LOC", "ORG"] and self.entity_prob_ind is a dict {"PER", [1, 2], ...}

File: deeppavlov/models/kbqa/entity_linking.py [hunk 15]
```diff
@@ -12,318 +12,332 @@
...
 
-import re
-import sqlite3
 from logging import getLogger
-from typing import List, Dict, Tuple, Optional, Any
+from typing import List, Dict, Tuple
 from collections import defaultdict
 
+import numpy as np
 import nltk
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:20]: redundant

File: deeppavlov/models/kbqa/entity_linking.py [hunk 16]
```diff
@@ -12,318 +12,332 @@
...
+        nums_batch = []
+        count_texts = 0
+        text = ""
+        curr_doc = 0
+        for n, doc in enumerate(docs_batch):
+            sentences = sent_tokenize(doc)
+            for sentence in sentences:
+                if len(text) + len(sentence) < self.max_chunk_len and n == curr_doc:
+                    text += sentence
+                    text += " "
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:63]: `text += f"{sentence} "` or `text += sentence + " "`

File: deeppavlov/models/kbqa/entity_linking.py [hunk 17]
```diff
@@ -12,318 +12,332 @@
...
+                else:
+                    if count_texts < self.batch_size:
+                        text_batch.append(text.strip(" "))
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:66]: `str.strip()` (without args) removes all space symbols (including tabulation)

File: deeppavlov/models/kbqa/entity_linking.py [hunk 18]
```diff
@@ -12,318 +12,332 @@
...
+                        if n == curr_doc:
+                            nums_batch.append(n)
+                        else:
+                            nums_batch.append(n-1)
+                        count_texts  += 1
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:71]:
```suggestion
                        count_texts += 1
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 19]
```diff
@@ -12,318 +12,332 @@
...
+    """
+        Class to split documents into chunks of max_chunk_len symbols so that the length will not exceed
+        maximal sequence length to feed into BERT
+    """
+
+    def __init__(self, max_chunk_len: int = 300, batch_size: int = 30, **kwargs):
+        self.max_chunk_len = max_chunk_len
+        self.batch_size = batch_size
+
+    def __call__(self, docs_batch: List[str]) -> Tuple[List[List[str]], List[List[int]]]:
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:50]: please, add docstring with arguments and response description

File: deeppavlov/models/kbqa/entity_linking.py [hunk 20]
```diff
@@ -12,318 +12,332 @@
...
+        entity_ids_batch_list = []
+        for text_batch in text_batch_list:
+            entity_ids_batch = []
+            ner_tokens_batch, ner_probas_batch = self.ner(text_batch)
+            entity_substr_batch, _, entity_positions_batch = self.ner_parser(ner_tokens_batch, ner_probas_batch)
+            log.debug(f"entity_substr_batch {entity_substr_batch}")
+            log.debug(f"entity_positions_batch {entity_positions_batch}")
+            entity_substr_batch = [[entity_substr.lower() for tag, entity_substr_list in entity_substr_dict.items()
+                                                         for entity_substr in entity_substr_list]
+                                                         for entity_substr_dict in entity_substr_batch]
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:198]:
```suggestion
                                    for entity_substr in entity_substr_list]
                                   for entity_substr_dict in entity_substr_batch]
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 21]
```diff
@@ -12,318 +12,332 @@
...
+            entity_positions_batch = [[entity_positions for tag, entity_positions_list in entity_positions_dict.items()
+                                                         for entity_positions in entity_positions_list]
+                                                         for entity_positions_dict in entity_positions_batch]
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:201]:
```suggestion
                                       for entity_positions in entity_positions_list]
                                      for entity_positions_dict in entity_positions_batch]
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 22]
```diff
@@ -12,318 +12,332 @@
...
+            log.debug(f"entity_substr_batch {entity_substr_batch}")
+            log.debug(f"entity_positions_batch {entity_positions_batch}")
+            for entity_substr_list, entity_positions_list, context_tokens in \
+                                               zip(entity_substr_batch, entity_positions_batch, ner_tokens_batch):
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:205]:
```suggestion
                    zip(entity_substr_batch, entity_positions_batch, ner_tokens_batch):
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 23]
```diff
@@ -12,318 +12,332 @@
...
+                else:
+                    doc_entity_ids_batch.append(doc_entity_ids)
+                    doc_entity_ids = entity_ids
+                    cur_doc_num = doc_num
+        doc_entity_ids_batch.append(doc_entity_ids)
+
+        return doc_entity_ids_batch
+
+    def link_entities(self, entity_substr_list: List[str], entity_positions_list: List[List[int]] = None,
+                          context_tokens: List[str] = None) -> List[List[str]]:
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:228]:
```suggestion
                      context_tokens: List[str] = None) -> List[List[str]]:
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 24]
```diff
@@ -12,318 +12,332 @@
...
+        log.debug(f"context_tokens {context_tokens}")
+        log.debug(f"entity substr list {entity_substr_list}")
+        log.debug(f"entity positions list {entity_positions_list}")
+        entity_ids_list = []
+        if entity_substr_list:
+            entity_substr_list = [[word for word in entity_substr.split(' ')
+                                        if word not in self.stopwords and len(word) > 0]
+                                        for entity_substr in entity_substr_list]
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:236]:
```suggestion
                                   if word not in self.stopwords and len(word) > 0]
                                  for entity_substr in entity_substr_list]
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 25]
```diff
@@ -12,318 +12,332 @@
...
+            words_and_indices  = [(self.morph_parse(word), i) for i, entity_substr in enumerate(entity_substr_list)
+                                                                                         for word in entity_substr]
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:238]:
```suggestion
                                 for word in entity_substr]
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 26]
```diff
@@ -12,318 +12,332 @@
...
+                          context_tokens: List[str] = None) -> List[List[str]]:
+        log.debug(f"context_tokens {context_tokens}")
+        log.debug(f"entity substr list {entity_substr_list}")
+        log.debug(f"entity positions list {entity_positions_list}")
+        entity_ids_list = []
+        if entity_substr_list:
+            entity_substr_list = [[word for word in entity_substr.split(' ')
+                                        if word not in self.stopwords and len(word) > 0]
+                                        for entity_substr in entity_substr_list]
+            words_and_indices  = [(self.morph_parse(word), i) for i, entity_substr in enumerate(entity_substr_list)
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:237]:
```suggestion
            words_and_indices = [(self.morph_parse(word), i) for i, entity_substr in enumerate(entity_substr_list)
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 27]
```diff
@@ -12,318 +12,332 @@
...
+            else:
+                entities_with_scores[entity] = score
+        entities_with_scores = list(entities_with_scores.items())
+        
+        return entities_with_scores
+
+    def rank_by_description(self, entity_positions_list: List[List[int]],
+                                  candidate_entities_list: List[List[str]],
+                                  entities_scores_list: List[Dict[str, Tuple[int, float]]],
+                                  context_tokens: List[str]) -> List[List[str]]:
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:317]:
```suggestion
    def rank_by_description(self, entity_positions_list: List[List[int]],
                            candidate_entities_list: List[List[str]],
                            entities_scores_list: List[Dict[str, Tuple[int, float]]],
                            context_tokens: List[str]) -> List[List[str]]:
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 28]
```diff
@@ -12,318 +12,332 @@
...
+                context = ' '.join(context_tokens[:entity_pos[0]]+["[ENT]"]+
+                                                   context_tokens[entity_pos[0]:entity_pos[-1]+1]+["[ENT]"]+
+                                                   context_tokens[entity_pos[-1]+1:])
+            else:
+                context = ' '.join(context_tokens[:entity_pos[0]]+["[ENT]"] + context_tokens[entity_pos[-1]+1:])
+            log.debug(f"context {context}")
+            log.debug(f"len candidate entities {len(candidate_entities)}")
+            scores = self.entity_ranker.rank_rels(context, candidate_entities)
+            entities_with_scores = [(entity, round(entities_scores[entity][0], 2), entities_scores[entity][1],
+                                     round(score,2)) for entity, score in scores]
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:333]:
```suggestion
                                     round(score, 2)) for entity, score in scores]
```

File: deeppavlov/models/kbqa/entity_linking.py [hunk 29]
```diff
@@ -12,318 +12,332 @@
...
+                                  context_tokens: List[str]) -> List[List[str]]:
+        entity_ids_list = []
+        for entity_pos, candidate_entities, entities_scores in zip(entity_positions_list, candidate_entities_list,
+                                                                   entities_scores_list):
+            log.debug(f"entity_pos {entity_pos}")
+            log.debug(f"candidate_entities {candidate_entities[:10]}")
+            if self.include_mention:
+                context = ' '.join(context_tokens[:entity_pos[0]]+["[ENT]"]+
+                                                   context_tokens[entity_pos[0]:entity_pos[-1]+1]+["[ENT]"]+
+                                                   context_tokens[entity_pos[-1]+1:])
```
Comment (IgnatovFedor) [inline deeppavlov/models/kbqa/entity_linking.py:326]:
```suggestion
                context = ' '.join(context_tokens[:entity_pos[0]] + ["[ENT]"] +
                                   context_tokens[entity_pos[0]:entity_pos[-1]+1] + ["[ENT]"] +
                                   context_tokens[entity_pos[-1]+1:])
```

File: docs/features/models/entity_linking.rst [hunk 30]
```diff
@@ -0,0 +1,37 @@
+Entity Linking
+========================================
+
+Entity linking is the task of mapping words from text (e.g. names of persons, locations and organizations) to entities from the target knowledge base (Wikidata in our case).
+
+Entity Linking component consists of the following steps:
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:6]:
```suggestion
Entity Linking component performs the following steps:
```

File: docs/features/models/entity_linking.rst [hunk 31]
```diff
@@ -0,0 +1,37 @@
...
+
+* the substring, detected with :config:`NER <ner/ner_rus_bert_probas.json>`, is fed to TfidfVectorizer and the resulting sparse vector is converted to dense one
+* `Faiss <https://github.com/facebookresearch/faiss>`__ library is used to find k nearest neighbours for tf-idf vector in the matrix where rows correspond to tf-idf vectors of words in entity titles
+* Entities are ranked by number of relations in Wikidata (number of outgoing edges of nodes in the knowledge graph)
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:10]:
```suggestion
* entities are ranked by number of relations in Wikidata (number of outgoing edges of nodes in the knowledge graph)
```

File: docs/features/models/entity_linking.rst [hunk 32]
```diff
@@ -0,0 +1,37 @@
...
+* :config:`BERT <classifiers/entity_ranking_bert_rus_no_mention.json>` is used for entities ranking by entity description and the sentence with the mention of the entity.
```
Comment (IgnatovFedor) [inline docs/features/models/entity_linking.rst:11]:
```suggestion
* :config:`BERT <classifiers/entity_ranking_bert_rus_no_mention.json>` is used for entities ranking by entity description and by sentence that mentions the entity
```

File: deeppavlov/models/kbqa/entity_detection_parser.py [hunk 33]
```diff
@@ -13,9 +13,11 @@
 # limitations under the License.
```
Comment (theanhle) [inline deeppavlov/models/kbqa/entity_detection_parser.py:13]: The full copyright and license should be added.

File: docs/features/models/entity_linking.rst [hunk 34]
```diff
@@ -0,0 +1,37 @@
...
+
+Entity linking is the task of mapping words from text (e.g. names of persons, locations and organizations) to entities from the target knowledge base (Wikidata in our case).
+
+Entity Linking component performs the following steps:
+
+* the substring, detected with :config:`NER <ner/ner_rus_bert_probas.json>`, is fed to TfidfVectorizer and the resulting sparse vector is converted to dense one
+* `Faiss <https://github.com/facebookresearch/faiss>`__ library is used to find k nearest neighbours for tf-idf vector in the matrix where rows correspond to tf-idf vectors of words in entity titles
+* entities are ranked by number of relations in Wikidata (number of outgoing edges of nodes in the knowledge graph)
+* :config:`BERT <classifiers/entity_ranking_bert_rus_no_mention.json>` is used for entities ranking by entity description and by sentence that mentions the entity
+
```
Comment (theanhle) [inline docs/features/models/entity_linking.rst:12]: - The model architecture should be detailed.

- It's better to show some results and compare the model performance with some other published models.

File: deeppavlov/core/common/registry.json [hunk 35]
```diff
@@ -95,6 +96,7 @@
   "nemo_tts": "deeppavlov.models.nemo.tts:NeMoTTS",
   "ner": "deeppavlov.models.ner.network:NerNetwork",
   "ner_bio_converter": "deeppavlov.models.ner.bio:BIOMarkupRestorer",
+  "ner_chunker": "deeppavlov.models.kbqa.entity_linking:NerChunker",
```
Comment (theanhle) [inline deeppavlov/core/common/registry.json:99]: What is the difference between `ner-chunker` and `chunk_finder` in `deeppavlov/metrics/fmeasure.py`?

File: deeppavlov/models/kbqa/entity_detection_parser.py [hunk 36]
```diff
@@ -50,45 +79,47 @@ def __call__(self, question_tokens: List[List[str]],
...
+            if tag_num in self.et_prob_ind:
                 if proba[tag_num] < self.thres_proba:
                     tag_num = 0
-            tags.append(tag_list[tag_num])
+            tags.append(self.tag_ind_dict[tag_num])
             tag_probas.append(proba[tag_num])
                     
         return tags, tag_probas
 
     def entities_from_tags(self, tokens, tags, tag_probas):
```
Comment (theanhle) [inline deeppavlov/models/kbqa/entity_detection_parser.py:94]: Please add the function description and arguments as well.

File: deeppavlov/core/common/registry.json (hunk 35 above)
Comment (dmitrijeuseew) [reply deeppavlov/core/common/registry.json:99]: ner_chunker splits a batch of documents into a batch of chunk lists (if the document is longer than the maximum sequence length for BERT, it is split into pieces).

File: deeppavlov/models/kbqa/entity_detection_parser.py (hunk 33 above)
Comment (dmitrijeuseew) [reply deeppavlov/models/kbqa/entity_detection_parser.py:13]: I copied copyright and license strings from other component files in our library.

File: deeppavlov/models/kbqa/entity_detection_parser.py (hunk 36 above)
Comment (dmitrijeuseew) [reply deeppavlov/models/kbqa/entity_detection_parser.py:108]: Added docstrings

File: deeppavlov/models/kbqa/entity_linking.py (hunk 4 above)
Comment (dmitrijeuseew) [reply deeppavlov/models/kbqa/entity_linking.py:20]: I mixed up nltk with numpy and wrongly deleted numpy in one of commits

File: docs/features/models/entity_linking.rst (hunk 34 above)
Comment (dmitrijeuseew) [reply docs/features/models/entity_linking.rst:12]: I do not know what dataset for entity linking in russian to use for comparison.

**Pull request merged**